art/utils.py

```diff
@@ -1096,7 +1096,7 @@ def preprocess(
     return normalized_x, categorical_y
 
 
-def segment_by_class(data: np.ndarray, classes: np.ndarray, num_classes: int) -> List[np.ndarray]:
+def segment_by_class(data: Union[np.ndarray, List[int]], classes: np.ndarray, num_classes: int) -> List[np.ndarray]:
```

Comment:
Why do is list of integer type required? Would it be possible to convert the list input into a ndarray?

Reply:
The `Union` operator won't require that the input is a list, but it will allow `List[int]` as well as `np.ndarray` in this function.

Here is how the `data` parameter is used in this function:
```python
by_class[assigned].append(data[indx])
```

The way the `data` argument is used in this function operates the same for `ndarray` and lists. Given this is a convenience utility method I think it's okay to accept both if that suits the use case.